src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -469,69 +468,78 @@ public FSDataInputStream getObject(String hostName, Path path) throws IOExceptio
 
     LOG.debug("List container for {} container {}", obj, container);
     ArrayList<FileStatus> tmpResult = new ArrayList<FileStatus>();
-    PaginationMap paginationMap = cObj.getPaginationMap(obj, pageListSize);
     FileStatus fs = null;
-    StoredObject previousElement = null;
-    for (Integer page = 0; page < paginationMap.getNumberOfPages(); page++) {
-      Collection<StoredObject> res = cObj.list(paginationMap, page);
-      if (page == 0 && (res == null || res.isEmpty())) {
-        FileStatus[] emptyRes = {};
-        LOG.debug("List {} in container {} is empty", obj, container);
-        return emptyRes;
+    SwiftObject previousElement = null;
+
+    Collection<SwiftObject> listing = SwiftAPIDirect.listContainer(mJossAccount,
+            swiftConnectionManager, container, obj);
+
+    if (listing.isEmpty()) {
+      FileStatus[] emptyRes = {};
+      LOG.debug("List {} in container {} is empty", obj, container);
+      return emptyRes;
+    }
+    for (SwiftObject object : listing) {
+      if (previousElement == null) {
+        // first entry
+        //setCorrectSize(tmp, cObj);
+        previousElement = object;
+        continue;
       }
-      for (StoredObject tmp : res) {
-        if (previousElement == null) {
-          // first entry
-          setCorrectSize(tmp, cObj);
-          previousElement = tmp.getAsObject();
```

Comment:
@gilv The logic is still there starting at line 483, it just shows up as removed because I had to change the indentation.